e2e/upgrade-cephfs.go

```diff
@@ -129,12 +149,45 @@ var _ = Describe("CephFS Upgrade Testing", func() {
 					e2elog.Failf("failed to load application with error %v", err)
 				}
 				app.Namespace = f.UniqueName
-				app.Labels = map[string]string{"app": "upgrade-testing"}
+				app.Labels = map[string]string{"app": "cephfs-upgrade-testing"}
+				pvc.Namespace = f.UniqueName
 				pvc.Spec.Resources.Requests[v1.ResourceStorage] = resource.MustParse(pvcSize)
 				err = createPVCAndApp("", f, pvc, app, deployTimeout)
 				if err != nil {
 					e2elog.Failf("failed to create pvc and application with error %v", err)
 				}
+				opt := metav1.ListOptions{
+					LabelSelector: "app=cephfs-upgrade-testing",
+				}
+				// fetch the path where volume is mounted.
+				mountPath := getMountPath(app)
+				filePath := filepath.Join(mountPath, "testClone")
+
+				// create a test file at the mountPath.
+				cmd := fmt.Sprintf("touch %s", filePath)
+				_, e := execCommandInPodAndAllowFail(f, cmd, app.Namespace, &opt)
+				if e != "" {
+					e2elog.Failf("failed to create file %s", e)
+				}
+
+				e2elog.Logf("Calculating checksum of %s", filePath)
+				checkSum, err = calculateSHA512sum(f, app, filePath, &opt)
+				if err != nil {
+					e2elog.Failf("failed to calculate checksum of %s", filePath)
+				}
+
+				// Create snapshot of the pvc
+				snapshotPath := cephfsExamplePath + "snapshot.yaml"
+				snap := getSnapshot(snapshotPath)
+				snap.Name = "cephfs-pvc-snapshot"
+				snap.Namespace = f.UniqueName
+				snap.Spec.Source.PersistentVolumeClaimName = &pvc.Name
+				// var s v1beta1.VolumeSnapshot
+				err = createSnapshot(&snap, deployTimeout)
+				if err != nil {
+					e2elog.Logf("failed to create snapshot %v", err)
+					Fail(err.Error())
+				}
```

Comment:
create a clone in this test itself

Reply:
The clone needs to be made after upgrading csi. Also, Clone is created from the snapshot and from PVC in separate stages for a clear understanding of different use cases.